dashboard/app/models/user_professional_learning_course_enrollment.rb

```diff
@@ -16,21 +16,22 @@
 class UserProfessionalLearningCourseEnrollment < ActiveRecord::Base
   belongs_to :user
   belongs_to :professional_learning_course
-  has_many :user_enrollment_module_assignment
+  has_many :user_enrollment_module_assignment, dependent: :destroy
   has_many :user_module_task_assignment, through: :user_enrollment_module_assignment
 
-  def self.enroll_user_in_course_with_learning_modules(user, course, learning_modules)
-    enrollment = UserProfessionalLearningCourseEnrollment.find_or_create_by(user: user, professional_learning_course: course, status: :in_progress)
+  def enroll_user_in_course_with_learning_modules(learning_modules)
+    self.user_enrollment_module_assignment.destroy_all
 
     learning_modules.each do |learning_module|
-      module_assignment = UserEnrollmentModuleAssignment.find_or_create_by(user_professional_learning_course_enrollment: enrollment, professional_learning_module: learning_module)
+      module_assignment = UserEnrollmentModuleAssignment.find_or_create_by(user_professional_learning_course_enrollment: self, professional_learning_module: learning_module)
 
       learning_module.professional_learning_tasks.each do |task|
         UserModuleTaskAssignment.find_or_create_by(user_enrollment_module_assignment: module_assignment, professional_learning_task: task, status: :not_started)
       end
     end
 
-    enrollment
+    self.status = :in_progress
```

Comment:
Not yet. I don't know what the final state is going to be, so I'm holding off for now.